### PR TITLE
assert.h: Don't use UNUSED macro since it's very easy happen conflict

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Improve the compatibility of assert

## Impact
Minor change

## Testing
Pass CI

### CODE REVIEW

File: include/assert.h [hunk 1]
```diff
@@ -67,7 +67,7 @@
  */
 
 #ifdef NDEBUG
-#  define assert(f) UNUSED(f)
+#  define assert(f)  ((void)(1 || (f)))
```
Comment (pkarashchenko) [inline include/assert.h:70]:
```suggestion
#  define assert(f) ((void)(1 || (f)))
```

**Comment by pkarashchenko** [comment]
I see that `#  define UNUSED(a) ((void)(1 || (a)))`, so why this change is needed?

**Comment by xiaoxiang781216** [comment]
> I see that `# define UNUSED(a) ((void)(1 || (a)))`, so why this change is needed?

UNUSED is frequently redefined by many third party to something like this:
```
# define UNUSED __attribute__((unused))
```
Since assert(assert.h) is the most used macro, it's better to remove the dependence on UNUSED macro.

File: include/assert.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/assert.h]: Done.

**Comment by pkarashchenko** [comment]
But `assert.h` includes `#include <nuttx/compiler.h>` and `UNUSED` is defined in `compiler.h` so the redefinition problem will still exists. I mean the problem will not be solved globally.

**Comment by pkarashchenko** [comment]
Maybe better to rename `UNUSED` to `__UNUSED` or something like this?

**Comment by pkarashchenko** [comment]
It's a chicken-egg problem. We meet similar with OpenSBI integration. Really do not know what is the best solution.

**Comment by xiaoxiang781216** [comment]
> But `assert.h` includes `#include <nuttx/compiler.h>` and `UNUSED` is defined in `compiler.h` so the redefinition problem will still exists. I mean the problem will not be solved globally.

the third party code write like this:
```
#undef UNUSED
#define UNUSED __attribute__((unused))
```

**Comment by pkarashchenko** [comment]
Yeah, but that is include order dependant.
We can merge this change if that solves your issue, but in general I'm against some partial solutions.

**Comment by pkarashchenko** [comment]
The another option is to add `#undef UNUSED` at the beginning of `nuttx/compiler.h`

**Comment by xiaoxiang781216** [comment]
> The another option is to add `#undef UNUSED` at the beginning of `nuttx/compiler.h`

It only work when we include the third party header file first and then assert.h, but the order is reverse in our case, that's why I create this patch. Since UNUSED is normally used inside source file. assert.h is the only exception by a recent change:
https://github.com/apache/incubator-nuttx/commit/5f3a98b5a8d39a7a9ee1823c6f7eaf05b1822ad7 
It's always good to avoid reference UNUSED in header files.

**Comment by pkarashchenko** [comment]
@xiaoxiang781216 I think that we can drop dependency on `nuttx/compiler.h` now.

**Pull request merged**